### ISSUE TITLE
Send telemetry from the browser in AspireWithNode sample

### ISSUE THREAD

**Issue by JamesNK** [dotnet/aspire-samples#409]
See https://github.com/dotnet/aspire/tree/main/playground/BrowserTelemetry as an example of how to do this.

**Comment by DamianEdwards** [comment]
I'm thinking we should just update the existing AspireWithJavaScript sample to do this. It has Angular, React, and Vue apps in it so seems like a good candidate to add this functionality.

/Cc @IEvangelist

**Comment by IEvangelist** [comment]
> I'm thinking we should just update the existing AspireWithJavaScript sample to do this. It has Angular, React, and Vue apps in it so seems like a good candidate to add this functionality.
> 
> /Cc [@IEvangelist](https://github.com/IEvangelist)

Tried doing this, only to learn that it's not really well supported. I spoke with @vhvb1989 and @davidfowl on this, and here's the conclusion as to why we cannot do this with the `AspireWithJavaScript` sample:

- Since these apps are SPAs, they're built in the container and their bits are pushed to a `/dist` folder and served by `nginx` as static files.
- In order to get the values of the `OTEL_EXPORTER_OTLP_ENDPOINT`, `OTEL_EXPORTER_OTLP_HEADERS`, and `OTEL_RESOURCE_ATTRIBUTES` at deployment time, I attempted to use the `PublishAsDockerFile` with the following build args:

    ```cs
    DockerBuildArg[] otelBuildArgs = [
        new("OTEL_EXPORTER_OTLP_ENDPOINT"),
        new("OTEL_EXPORTER_OTLP_HEADERS"),
        new("OTEL_RESOURCE_ATTRIBUTES")
    ];
    ```
    
    However, these env vars are not available during build when publishing the app.

- At runtime, these env var values are available, but at that point - it's too late.
- Also, ACA doesn't expose the OTEL collector.

**Comment by DamianEdwards** [comment]
@IEvangelist everything you're referring to is regarding supporting this after publishing to ACA, correct? IIUC we could make the sample support it at dev time without much trouble but it wouldn't work after publishing, right?

**Comment by IEvangelist** [comment]
Yeah, again I have this all working locally and correctly reporting browser telemetry for each SPA. I could add a note in the README.md to say something about the deployment not working. Perhaps, we should remove the Dockerfiles from the sample then too, as it's really confusing to include them since they will not work.

**Comment by DamianEdwards** [comment]
Could we update the sample logic so that the client app gets the URL via a call to a backend API?

**Comment by ethangillengg** [comment]
Hi all, I am trying to implement this for an aspire app using Vite as the SPA bundler. I am also in the same situation where OTEL from the browser is working in dev, but I cannot get it working on my deployment ACA.

@IEvangelist you mentioned that:
> - Also, ACA doesn't expose the OTEL collector.

Would this mean that there is no way currently to get OTEL from the browser when deploying to ACA? Thanks!
